src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -47,6 +47,21 @@ function * fetchCategory (dataContainer: DataContainer, action: FetchCategoryAct
   }
 }
 
+function * cancelableFetchCategory (dataContainer: DataContainer, action: FetchCategoryActionType): Saga<void> {
+  const { cancel } = yield race({
+    response: call(fetchCategory, dataContainer, action),
+    cancel: take('SWITCH_CONTENT_LANGUAGE')
+  })
+
+  if (cancel) {
+    const newFetchCategory = action
+    const newLanguage = cancel.params.newLanguage
+    newFetchCategory.params.language = newLanguage
+    newFetchCategory.params.path = `/${action.params.city}/${newLanguage}`
+    yield put(newFetchCategory)
```

Comment:
Create a new action here and copy the relevant stuff over. Immutable data ftw :)